cdslib/agent/movement.py

```diff
@@ -0,0 +1,30 @@
+from pandas.core.frame import DataFrame
+
+from cdslib.models.population import BoxSize
+
+
+class AgentMovement:
+
+    @classmethod
+    def apply_movement(cls, dataframe: DataFrame, box_size: BoxSize):
+        """
+
+        """
+        dataframe.x += dataframe.vx
+        dataframe.y += dataframe.vy
+
+        if dataframe.x < box_size.left:
+            dataframe.vx = -dataframe.vx
+            dataframe.x = box_size.left
+        if dataframe.x > box_size.right:
+            dataframe.vx = -dataframe.vx
+            dataframe.x = box_size.right
+
+        if dataframe.y < box_size.bottom:
+            dataframe.vy = -dataframe.vy
+            dataframe.y = box_size.bottom
+        if dataframe.y > box_size.top:
+            dataframe.vy = -dataframe.vy
+            dataframe.y = box_size.top
+
+        return dataframe
```

Comment:
It is supposed that dt must appear as a parameter of this functions, thus changing code lines 13 and 14: multiply by dt.

In addition, try to use df instead of dataframe.

Reply:
### Solved
Add dt as parameter in the function, and multiply velocities in the agent